### DISCUSSION TITLE
Trying to use guidance in Colab environment

### DISCUSSION THREAD

**Discussion by taehalim** [guidance-ai/guidance#261] [Q&A] [answered]
I imported `guidance` on Colab just fine but when I execute the following code an error pops:
```python
program = guidance('''The best thing about the beach is {{~gen 'best' temperature=0.7 max_tokens=7}}''')
program()
```
The Error:
```
AssertionError: You must provide an OpenAI API key to use the OpenAI LLM. Either pass it in the constructor, set the OPENAI_API_KEY environment variable, or create the file ~/.openai_api_key with your key in it.
```
So I added the following (tried to add environment variable):
```python
import os
os.environ['OPENAI_API_KEY'] = "MY_API_KEY"
```
But, I still had the same issue. Anyone has any idea why this isn't working??? I'm also curious how you guys are using guidance apart from using it on Colab. I believe there are few steps I need to setup before using guidance properly, for example, there is specific document in LangChain on how to use it from scratch. ANY helpful info would be nice. Thanks you so much.

**Comment by taehalim** [comment, marked as answer]
Solved it by importing openai package but I'm still confused. 
```python
import openai
openai.api_key = "YOUR_API_KEY"
```
I thought `os.environ['OPENAI_API_KEY'] = "MY_API_KEY"` would also work 😢

**Reply by rickbraddy-pharma** [reply]
That does work. We use Colab too. We place a constants.py file in the root or other folder on Google Drive. Then import and use the API keys or other constants that vary across runtime environments… works for us.

**Reply by taehalim** [reply]
Thanks. I'll try your approach. appreciate it !
